core/commands/storage.go

```diff
@@ -794,7 +796,13 @@ the shard and replies back to client for the next challenge step.`,
 		}
 		// build session
 		sm := storage.GlobalSession
-		ss := sm.GetOrDefault(ssID, n.Identity)
+		ss, err := sm.GetSession(n, storage.ShardsStorePrefix, ssID)
+		if err != nil {
+			log.Error(err)
+		}
+		if ss == nil {
+			ss = sm.GetOrDefault(ssID, n.Identity)
```

Comment:
This is verbose, you can refactor it later: it should just be part of `GetSession` to generate default.